furusiya/model/keys/decorators.py

```diff
@@ -17,6 +17,7 @@ def _inner_function(event):
             if Game.game_state == 'playing' and Game.current_turn is Game.player:
                 if pass_turn:
                     Game.current_turn = None
+                    Game.events.trigger('on_turn_pass')
```

Comment:
I say this shouldn't be there. It would probably be better placed in UpdateManager's `update` method.

Reply:
I looked at that method and didn't see a clear indication of where it should go. My guess is that `update` is called regardless of who's turn; and if it's not the player's turn, everything else gets a turn, correct?  If so, I might place it under the `else` clause.  But would that work with, say, resting, or consuming items? These things take time.

The current implementation does fizzle out fire when you rest (I tested it).

Reply:
I would very much prefer it to be in `update`, but this works, so it's Good Enough for now.